### PR TITLE
feat(README): example sections with services, controllers, directives

### PR DESCRIPTION
**Pull request by tykowale**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -378,4 +383,76 @@ This is helpful to test in IE 9 and IE 10.
...
+beforeEach(inject(function (_$controller_) {
+    $controller = _$controller_;
+
+    stateParams = {
+        awesomeMethod: sinon.stub() 
+    };
+}));
+
+function init() {
+    SampleController = $controller('SampleController', {
```
Comment (alexlrobertson) [inline README.md]: `Uncaught ReferenceError: SampleController is not defined.`

File: README.md [hunk 2]
```diff
@@ -378,4 +383,76 @@ This is helpful to test in IE 9 and IE 10.
...
+## Examples
+### Services
+
+Testing a service is as easy as injecting a dependency. Keep naming consistent to improve readability of tests.
+ 
+```javascript
+    
+beforeEach(module('up.moduleServiceIsIn'));
+beforeEach(module(function ($provide) {
+    upNamedService1 = {
```
Comment (alexlrobertson) [inline README.md]: Can we use real service names to clarify the difference between them?

Comment (alexlrobertson) [inline README.md]: Numbers on the end of variables is a code smell. :nose:

File: README.md [hunk 3]
```diff
@@ -378,4 +383,76 @@ This is helpful to test in IE 9 and IE 10.
...
+```javascript
+beforeEach(module('up.sampleModule'));
+beforeEach(module(function ($provide) {
+    $state = {
+        get: sinon.stub()
+    };
+    $provide.value('$state', $state);
+}));
+beforeEach(inject(function(_$compile_, $rootScope) {
+    $rootScope = $rootScope.$new();
```
Comment (alexlrobertson) [inline README.md]: `scope = $rootScope.$new();`

We shouldn't be reassigning to `$rootScope`.

File: README.md [hunk 4]
```diff
@@ -378,4 +383,76 @@ This is helpful to test in IE 9 and IE 10.
...
+    $compile = _$compile_;
+}));
+
+function compile() {
+    html = $compile(angular.element('<up-directive></up-directive>'))($rootScope);
+    $rootScope.$apply();
+    isolateScope = html.isolateScope();
+    }
```
Comment (alexlrobertson) [inline README.md]: :hankey: `\t`

File: README.md [hunk 5]
```diff
@@ -320,12 +325,12 @@ beforeEach(module(function ($provide) {
 }));
 ```
 
-#### Don’t use `$provide` when testing controllers
+### Use `$controller` when testing controllers
```
Comment (alexlrobertson) [inline README.md]: This probably shouldn't be bumped up a heading level (yet).

**Comment by alexlrobertson** [comment]
Can we show some sample expectations?

File: README.md (hunk 5 above)
Comment (tykowale) [inline README.md]: I was following the convention for `### Use `$provide` to mock services from other modules`. I feel like they are on the same level

Comment (alexlrobertson) [inline README.md]: But this is specifying more about `### Use $provideto mock services from other modules`...

**Comment by alexlrobertson** [comment]
:cake:

**Pull request merged**